lib/Basics/Guarded.h

```diff
@@ -174,8 +178,10 @@ class Guarded {
   using mutex_guard_type = MutexGuard<value_type, lock_type>;
   using const_mutex_guard_type = MutexGuard<value_type const, lock_type>;
 
-  auto getLockedGuard() -> mutex_guard_type;
-  auto getLockedGuard() const -> const_mutex_guard_type;
+  auto getLockedGuard(std::optional<basics::SourceLocation> = {})
+      -> mutex_guard_type;
+  auto getLockedGuard(std::optional<basics::SourceLocation> = {}) const
+      -> const_mutex_guard_type;
```

Comment:
I think the `SourceLocation` debug information here and in the UnshackledMutex should be removed. Or at least made available only in maintainer mode in a way that doesn't add costs without it, but I don't see an easy way to get rid of the parameter in that case.